### PR TITLE
Add unit tests of C++ static constructor initialization

### PR DESCRIPTION
**Pull request by Blackhex**
Also adds `.github/scripts/tests/execute-tests.sh` that can be used to execute the tests from WSL environment using interop.

I am not sure this would reproduce the former issue becase C linker is used (instead of C++ linker).

### CODE REVIEW

File: .github/scripts/tests/execute-tests.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (eukarpov) [inline .github/scripts/tests/execute-tests.sh:1]: I am not sure, for what this file is introduced.

File: tests/static-constructor.c [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include "gtest_like_c.h"
+
```
Comment (eukarpov) [inline tests/static-constructor.c:2]: I would keep only static-constructor.cpp.

File: .github/scripts/tests/execute-tests.sh (hunk 1 above)
Comment (Blackhex) [reply .github/scripts/tests/execute-tests.sh:1]: It allows executing the tests from WSL command line, including proper setting of host `PATH` so the toolchain DLLs are used.

File: tests/static-constructor.c (hunk 2 above)
Comment (Blackhex) [reply tests/static-constructor.c:2]: Non-static instance removed.

Comment (eukarpov) [reply tests/static-constructor.c:2]: I think static-constructor.c can be removed.

Comment (Blackhex) [reply tests/static-constructor.c:2]: Without the code in \`static-constructor.c\` the tests would not be included in the test suite, i.e. executed during the workflow run. Is that what you mean?

Comment (eukarpov) [reply tests/static-constructor.c:2]:
```
TEST(Aarch64MinGW, StaticConstructorTest)
{
    ASSERT_TRUE(static_construtor());
}
```
This code is present in cpp. Is it not enough?

Comment (Blackhex) [reply tests/static-constructor.c:2]: That code is compiled in with real GTest, I made it consistent e.g. with [mingw-woarm64-build/tests/sscanf-double.cpp](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/tests/sscanf-double.cpp) even if the real GTest is currently not used. The C implementation of GTest cannot be compiled with C++ compiler.

Comment (eukarpov) [reply tests/static-constructor.c:2]: Even after placing TEST(Aarch64MinGW, StaticConstructorTest) in extern "C"?

Comment (Blackhex) [reply tests/static-constructor.c]: Yes, that works. Thank you for the suggestion. I was also wrong with the assumption that the C++ file were including/using the real GTest implementation.

**Pull request merged**